feat: Implement Main page & Implement transitions with pixel animations and hover effects

rhino-ty opened the pull request:
## 📂 작업 내용

closes #15

- [x] 페이지 전환 애니메이션 및 로고 호버 효과 추가
- [x] `PixelTransitionContainer` 컴포넌트를 이용한 커스터마이징 가능한 페이드 아웃 전환 효과 구현
- [x] 메인 페이지 전환, 네비게이션 및 레이아웃 조정 작업 완료

## 💡 자세한 설명

- **PixelTransitionContainer 컴포넌트 추가**
  - 페이지 전환 시 픽셀 그리드 기반 슬라이드 아웃 애니메이션 적용, 애니메이션 방향(좌/우/상/하) 설정 가능. 
  - 스크린 리더를 위한 접근성 속성(ARIA)도 포함.
- **버튼 및 스타일 최적화**
  - 버튼의 변형명 오타를 수정(‘proimary’에서 ‘primary’로).
  - Button 컴포넌트의 기본 스타일을 단순화하고, 반응형 스타일링은 구현 레벨로 이동하여 일관성 유지.
- **TailwindCSS 플러그인 추가**
  - `tailwindcss-animate` 플러그인 사용해 애니메이션 효과 적용.
- **메인 페이지의 디자인 개선**
  - `setTimeout` + 임시 라우팅 로직을 구현하고, 페이지 전환에 픽셀 그리드 효과를 적용해 부드러운 전환을 제공.
  - 로고에 호버 애니메이션 추가.

## 📗 참고 자료 & 구현 결과 

https://github.com/user-attachments/assets/103eeb70-8cf1-4b38-bb4e-594ba7d8b0b6

나중에 사각형 안에 이미지 같은 걸 넣어도 될 듯?

## 📢 리뷰 요구 사항

- PixelTransitionContainer의 접근성 속성 검토 필요
- 페이지 전환 애니메이션의 방향 설정이 UX에 적합한지 확인

## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 테스트는 잘 통과했나요?
- [x] 불필요한 코드는 제거했나요?

rhino-ty committed:
fix: Correct typo in button variant name

- Fix spelling of 'proimary' to 'primary' in button variants

rhino-ty committed:
design: Simplify Button component base styles

- Remove responsive breakpoints from base button variants
- Simplify border styles by using consistent border-2
- Move responsive styling responsibility to implementation level

rhino-ty committed:
chore: Add tailwindcss-animate plugin and import statement

- Import tailwindcssAnimate from 'tailwindcss-animate'

rhino-ty committed:
feat: Add PixelTransitionContainer for page fade-out animations

- Implement customizable exit directions (left/right/up/down)
- Create pixel grid overlay with dynamic transition effects
- Add accessibility attributes for screen readers

- Smooth transition animations with pixel grid effects
- Configurable exit directions
- Responsive grid layout
- Proper z-index handling
- ARIA attributes for better accessibility

rhino-ty committed:
feat: Enhance main page with animations and transitions

- Add pixel transition effect for page navigation
- Implement hover animation for logo
- Add slide-out animations
- Setup navigation to waiting room
- Improve responsive layout with better spacing
- Add temporary routing logic with setTimeout

rhino-ty committed:
fix: Fix typo

choiseona commented on client/src/pages/MainPage.tsx:
```diff
@@ -1,14 +1,42 @@
+    // const { roomId } = response.data;
+
+    setIsExiting(true);
+    setTimeout(() => {
+      navigate('/waiting-room/123');
+    }, 1000);
```
여기서 setTimeout을 하는 이유가 궁금합니다!

rhino-ty replied:
`navigate`가 전환 애니메이션이 끝난 1초 뒤에 해야돼서 그래요,, 생각해보니 그냥 커스텀훅으로 분리하면 되겠네요

choiseona replied:
아니면 PixelTransitionContainer 컴포넌트에서 callback 함수 받아서 애니메이션이 끝나면(초를 직접 지정하거나 animationEnd 이벤트 발생시) callback 실행시키는 방법도 있을 것 같아요!

초를 직접 지정하시는거면 저의 timer util 사용 어떠신가요 ㅎ.ㅎ

choiseona replied:
PixelTransitionContainer 컴포넌트에서 callback 함수 받아서 애니메이션이 끝나면(초를 직접 지정하거나 animationEnd 같은 이벤트 발생시) callback 실행시키는 방법도 있을 것 같아요!

초를 직접 지정하시는거면 util에 timer도 있답니당

rhino-ty replied:
저 컴포넌트는 단일 책임 원칙에 의거해서 UI 렌더링만 해야하기 때문에 비즈니스 로직을 추가 안할 거에요! `애니메이션 끝난 후 실행 + 타이머` 같은 비즈니스 로직은 수행은 안하도록 할겁니다. 또 메인페이지와 컴포넌트 2개 다 필요한 상태가 있어서 공유할 상태가 필요하기 때문에 Hook으로 빼야할 거 같아요.

`util`에 `timer`도 고려했었는데 사용 안한 이유는

1. CSS로 만든 transition으로도 충분함
2. 내 컴포넌트와는 구조가 맞지 않아 복잡성 증대, 카운트다운, 자동 닫힘 등에 더 적합
3. CSS로 정의된 애니메이션 시간과 JS Timer가 불일치할 수 있음

이정도가 있겠네요! 저는 그냥 `setTimeout`만 지정하면 되니깐여

choiseona commented on client/src/pages/MainPage.tsx:
```diff
@@ -1,14 +1,42 @@
+        <div className="duration-1000 animate-in fade-in slide-in-from-top-8">
+          <Logo variant="main" className="w-full transition duration-300 hover:scale-110 hover:brightness-[1.12]" />
+        </div>
+
+        <Button onClick={handleCreateRoom} className="h-12 max-w-[280px] animate-pulse">
+          방 만들기
```
여기 max-w-[280px] 만 px로 돼있는데 이유가 있을까요?

rhino-ty replied:
없습니다 귀찮았기 때문이죠! 수정하려고 했는데 바보같으니!! 테일윈드 유틸리티로 바꾸겠습니다.

rhino-ty committed:
refactor: Extract page transition logic into custom hook

- Move transition animation logic from PixelTransitionContainer to usePageTransition hook

rhino-ty committed:
feat: Add eastbay color theme variant to PixelTransitionContainer

- Add colorTheme variant with violet and eastbay options
- Split variants into container and pixel styles
- Use data-exiting attribute for state-based styling
